hermes-common/src/main/java/pl/allegro/tech/hermes/common/hook/HooksHandler.java

```diff
@@ -21,24 +23,35 @@ public void addShutdownHook(ServiceAwareHook hook) {
     }
 
     public void shutdown(ServiceLocator serviceLocator) {
+        deregisterGlobalShutdownHook();
```

Comment:
But it looks like it is disabled for all use cases because its in common package...